src/Package/Fetch.zig

```diff
@@ -1099,7 +1099,12 @@ fn unpackResource(
 
     switch (file_type) {
         .tar => try unpackTarball(f, tmp_directory.handle, resource.reader()),
-        .@"tar.gz" => try unpackTarballCompressed(f, tmp_directory.handle, resource, std.compress.gzip),
+        .@"tar.gz" => {
+            const reader = resource.reader();
+            var br = std.io.bufferedReaderSize(std.crypto.tls.max_ciphertext_record_len, reader);
```

Comment:
Does your implementation actually require an additional buffered reader? This was here because the previous implementation did many small reads (1-8 bytes). If your implementation calls read() with a buffer size closer to 16K (the value of `max_ciphertext_record_len`) then this is not necessary.

Reply:
When I first test fetching without buffered reader it failed with EndOfStream so I just put it because it was in previous implementation. Now I check why that is necessary and found bug in reading input stream in decompression. I was using read instead of readAll and getting less bytes then available in the stream. Now when fixed buffered reader is not needed any more.

Decompression is doing many small reads, as in previous implementation. Internal bit reader has buffer of 8 bytes. Decompressor has output buffer of 64K (it needs history data of at least 32K) but there is no input buffer it only gets as many bytes as needed to fill 8 byte bit buffer. 

Compressor is other way around. It has 64K input buffer, reads in chunks of 32K and small output buffer of 248 bytes.

Currently I didn't remove buffered reader reasoning that behavior is same as previous.

Reply:
> Now I check why that is necessary and found bug in reading input stream in decompression. I was using read instead of readAll and getting less bytes then available in the stream. Now when fixed buffered reader is not needed any more.

Note that calling read() instead of readAll() is appropriate for implementation of a stream. This will block on the underlying read() call only once, allowing users of the stream to possibly do other things with the thread while waiting for more data to be available to read.

However, make sure to notice correctly when the end of stream occurs. A short read does not mean the end of stream.

Reply:
After writing the above comment, I looked at the commit you were referring to, and I see that in this case the amount read is very small (8 bytes or less). In this case it is OK to use readAll.

A potentially better solution would be to still use only 1 call to read(), and have an internal small buffer of those 8 bytes, and then only emit the bytes from read when the buffer is filled, however, this will require an updated definition of std.io.Reader.read because currently it says if the number of bytes read is 0 it means end of stream.